### ISSUE TITLE
create a grunt task to compare PDOM descriptions

### ISSUE THREAD

**Issue by zepumph** [phetsims/perennial#138]
From a11y dev meeting today with @jessegreenberg, we were interested in prototyping around with a way to test description comparisons between two versions. We worked on a prototype today, and got to a commit point where we can compare the PDOMs on startup of working copy sim with the same from master (using `git stash`ing). This is working, but there are some pretty big todos to think about before diving fully in. Let's talk more about this @jessegreenberg!

**Comment by jessegreenberg** [comment]
In #139 we also discovered that Ctrl+V adds extra `\u0016` characters that may be interfering with things.

**Comment by zepumph** [comment]
Notes from discussion:
```
usage:
cd gravity-force-lab
grunt pdom-comparison --sha=7584ghfjde758g905y4hbj

working copy vs 7584ghfjde758g905y4hbj

Algorithm:
get pdoms for current working copy
stash all repos
checkout 7584ghfjde758g905y4hbj
get pdoms for 7584ghfjde758g905y4hbj
restore copy

TODO how to get good PDOMs to compare (maybe compare PDOMs with deterministic fuzzing (multiple copies of the PDOM))
TODO how to display the results?
```

**Comment by jessegreenberg** [comment]
The notes in https://github.com/phetsims/perennial/issues/138#issuecomment-509760019 illustrate that we discussed comparing the PDOM against a given date by SHA. This way, we aren't locked into a single PDOM defined at publication time and we don't compare against master which would make it easy to miss things.

**Comment by zepumph** [comment]
@twant and I worked on this today. We were able to get very far on testing the current working copy (even with unstashed changes) against a previous sim commit. 

given this working copy change in `GFLBScreenView`: 
![image](https://user-images.githubusercontent.com/6856943/60931936-18ed1580-a268-11e9-96f9-a9b3f388f9ab.png)


And this command, where the sha is most recent commit in gravity-force-lab-basics:
`grunt pdom-comparison --debug --sha=5ffa6afc09fb1f625d65e16a16f7d3c6e54255b4`

We were able to see this output in git bash:
![image](https://user-images.githubusercontent.com/6856943/60931935-17bbe880-a268-11e9-86fe-d67321afc5d5.png)

The next steps are marked as todos in the file, but here is what I think should be worked on and noted:
1. this script needed to call CHIPPER/getPhetLibs, which is problematic because perennial should be dependencyless
2. We are still only comparing startup PDOMs, and it may be nice to compare with some fuzzing.
3. There was a fair bit of noise potential, and it is a bit hard to understand what the actual issues are, even from just a few days back. Here is an example from testing last week:
  ![image](https://user-images.githubusercontent.com/6856943/60932132-1808b380-a269-11e9-87df-f8101b0b06b3.png)

4. Use the `load` postMessage event for more efficient running of requirejs sims.

**Comment by zepumph** [comment]
This task was breaking a MR that @chrisklus was working on, because checkout-master-all was unable to load the chipper `getPhetLibs` script. To bypass it, I did the same workaround that we do to support lint-everything from perennial: requiring only within the grunt task. Sorry for the trouble! Tagging @jessegreenberg too.

**Comment by pixelzoom** [comment]
While working on https://github.com/phetsims/tasks/issues/1017, I discovered that this apparently has introduced a dependency on chipper in perennial:

```js
  // TODO: perennial shouldn't depend on chipper, https://github.com/phetsims/perennial/issues/138
  const dependencies = getPhetLibs( repo );
```

And this dependency has existed since 7/29/19.  That seems potentially very bad, and should be addressed immediately, no?  Assigning to @zepumph and labeling for developer meeting.

**Comment by zepumph** [comment]
This issue is for an experimental grunt task that isn't currently used. There is a dependencies getter in perennial that I was able to use instead of calling over to chipper. Thanks for the poke on the TODO.

**Comment by zepumph** [comment]
Keeping labelled for dev meeting in case @pixelzoom had other thoughts beyond fixing the issue in master.

**Comment by zepumph** [comment]
The dependency on chipper was removed above, after discussing in dev meeting, nothing else is needed to be done here for that problem.

**Comment by zepumph** [comment]
Over in friction, I'm about to tear up some PDOM descriptions to help support voicing. It would be just excellent if we could get something here working to support that effort. I would want to know if there were any problems I was creating earlier rather than later.

I think that my first idea above was a good one, but I didn't really know about the snapshot comparison tool, which is already pretty well set up to make comparisons with pixels. Perhaps I could add a module to it to also compare the PDOM HTML and perhaps some sort of serialization of alerts too (description and voicing while we're at it).

**Comment by zepumph** [comment]
This issue was first about a grunt task. I'd rather create a new issue for experimenting with snapshot comparison, so I'll pick this up in https://github.com/phetsims/aqua/issues/127. If things go well. I'll delete the grunt task and close this issue.

**Comment by zepumph** [comment]
https://github.com/phetsims/aqua/issues/127 is totally the way of the future. After giving @jessegreenberg a tour, I am ready to give up on this old strategy.

**Comment by zepumph** [comment]
Removed above. Closing
